DecipherNow/gm-fabric-dashboard
# Substantial refactor of Sidebar and Routing

Firstly, this PR solves the sidebar regressions introduced by a previous PR:
* Sidebar SummaryCard open and close their drawers to show detailed metrics when clicked
* Sidebars are highlighted as active based on the current route as expected

Additionally, this PR substantially modularizes the way that the app handles different runtimes. Runtime-specific components are organized into their own folders and have a much more sane and extensible structure. The sidebar has also been refactors into a clear hierarchy of components.

Resolves #382 

## To Test

To view JVM:
1. Open a terminal session to the project directory and run `npm run mock-jvm`
2. Open `/public/index.html` and ensure runtime is set to `JVM` on line 15 (it may already be set)
3. Open `package.json` and ensure that proxy is set to `http://localhost:9990`
4. Open a terminal session to the project directory and run `npm start`

You should see something like the following:
![image](https://user-images.githubusercontent.com/8700322/30060394-2ce42f10-9211-11e7-8a5c-a3114e6f4e86.png)

If you see fewer cards than expected, please "Reset Dashboards" from the settings page 

To make sure the bugs were fixed:
1. Click on the toggle on one of the sidebar cards. You should see a drawer with metrics open / close.
2. Click on different sidebar cards. The sidebar card should navigate you to a different route and the card should be styled as currently active.

To view Go: 
1. End all terminal processes used to show JVM
2. Open a terminal session to the project directory and run `npm run mock-go`
3. Open `/public/index.html` and ensure runtime is set to `GOLANG` on line 15
4. Open `package.json` and ensure that proxy is set to `http://localhost:9092`
5. Open a terminal session to the project directory and run `npm start`

You should see something like the following. Notice that this dashboard has Percent CPU utilized, but the JVM dashboard did not. This is because this is a new metric that was not offered by Finagle.
![image](https://user-images.githubusercontent.com/8700322/30060440-5a533d92-9211-11e7-83bc-10af9154d261.png)
